pkg/skaffold/initializer/init_test.go

```diff
@@ -431,3 +431,70 @@ func TestAutoSelectBuilders(t *testing.T) {
 		})
 	}
 }
+
+func TestProcessCliArtifacts(t *testing.T) {
+	tests := []struct {
+		description   string
+		artifacts     []string
+		shouldErr     bool
+		expectedPairs []builderImagePair
+	}{
+		{
+			description: "Invalid pairs",
+			artifacts:   []string{"invalid"},
+			shouldErr:   true,
+		},
+		{
+			description: "Invalid builder",
+			artifacts:   []string{`{"name":"Not real","payload":{}}=image`},
+			shouldErr:   true,
+		},
+		{
+			description: "Valid (backwards compatibility)",
+			artifacts: []string{
+				`/path/to/Dockerfile=image1`,
+				`/path/to/Dockerfile2=image2`,
+			},
+			expectedPairs: []builderImagePair{
+				{
+					Builder:   docker.Docker{File: "/path/to/Dockerfile"},
+					ImageName: "image1",
+				},
+				{
+					Builder:   docker.Docker{File: "/path/to/Dockerfile2"},
+					ImageName: "image2",
+				},
+			},
+		},
+		{
+			description: "Valid",
+			artifacts: []string{
+				`{"name":"Docker","payload":{"path":"/path/to/Dockerfile"}}=image1`,
+				`{"name":"Jib Gradle Plugin","payload":{"path":"/path/to/build.gradle"}}=image2`,
```

Comment:
What about a test where `name` is a valid one but `payload` is invalid?

Reply:
Current behavior is that `payload` is always valid (unknown fields are skipped, and missing expected fields just end up being `nil`), except for straight up invalid json. Do you think it should error if required fields are missing (although I'm not sure which fields would be considered "required")?

Reply:
Hang on, I might be wrong.

Reply:
Okay, I think this is fine. I think for both Docker and Jib builders, `path` should always exist, but I guess if it's missing, other parts of the codebase will error out appropriately.